include/aws/s3/s3_client.h

```diff
@@ -491,19 +492,100 @@ AWS_S3_API
 void aws_s3_meta_request_cancel(struct aws_s3_meta_request *meta_request);
 
 /**
- * In order to pause an ongoing upload, call aws_s3_meta_request_pause(). It will return a resume token that can be
- * persisted and used to resume the upload. To resume an upload that was paused, supply the resume token in the meta
- * request options structure member aws_s3_meta_request_options.persistable_state.
+ * Note: pause is currently only supported on upload requests.
+ * In order to pause an ongoing upload, call aws_s3_meta_request_pause() that
+ * will return resume token. Token can be used to query the state of operation
+ * at the pausing time. Use aws_byte_buf_init_from_s3_resume_token to serialize token.
+ * To resume an upload that was paused, supply the serialized resume token in the meta
+ * request options structure member aws_s3_meta_request_options.resume_token.
  * The upload can be resumed either from the same client or a different one.
- * Resume token is opaque with format varying based on operation.
- * Clients should not parse the token. For format details refer to pause method comments for a given operation.
  * Resume token will be set to null in case of failures.
+ * Note: similar to cancel pause does not cancel requests already in flight and
+ * and parts might complete after pause is requested.
+ * Note: pausing meta request before it starts will succeed with NULL token
+ * returned and will pause request. (resuming with null token is equivalent to
```

Comment:
this is really weird, to return AWS_OP_SUCCESS, but then the token is NULL

do you think users will want to treat not-yet-started requests differently than OTHER paused requests?

maybe just return a token that knows it was never really started, and there are functions the user can query to learn that fact.

or report a unique error like `AWS_ERROR_S3_REQUEST_NOT_YET_STARTED`, instead of reporting success but leaving the token NULL

Reply:
The not-yet-started case is very similar to the already-complete case.
Either case could come up when the user calls pause(), but one reports and ERROR and the other does not.

we should consider doing the same thing in both cases...

either both report a unique error
or both return a token, that the user can query to realize it's actually already complete, or actually it never really started.

Reply:
i actually had a separate error for it while i was playing around with it. 
my thinking on returning null is that null signifies basically that request should start from beginning, which is the case with pause where we haven't started processing it yet. but lets have a quick discussion about it on mon